src/visitor.js

```diff
@@ -144,6 +144,10 @@ class VisitState {
             path.node.body.unshift(T.expressionStatement(increment));
         } else if (path.isStatement()) {
             path.insertBefore(T.expressionStatement(increment));
+        } else if ((path.isFunctionExpression() || path.isArrowFunctionExpression()) && T.isVariableDeclarator(path.parentPath)) {
+            path.parentPath.parentPath.insertBefore(T.expressionStatement(
+                increment
```

Comment:
Normally I'd agree, but unfortunately the last branch of logic generates a `T.sequenceExpression`, rather than a `T.expressionStatement()`, so in many cases we'd actually be doing additional processing if we generated the `expressionStatement` node ahead of time.